### DISCUSSION TITLE
Mimics Copy Player Inputs at Point of Recording & Item Holding

### DISCUSSION THREAD

**Discussion by MattWojo73** [qwbarch/mirage#103] [Ideas]
I'd imagine this would be way more complicated than what this mod is intended to do, but I thought it'd be cool enough to throw out there.

If at the point the mod copy's a player's audio, it could also copy the inputs they're doing at that exact moment. Such as jumping, crouching, looking around, etc. To give a bit more believably to what that player was doing while they were saying it.

I understand that could be really complicated purely in the sense of maintaining the AIs pathing towards a player (and I don't even know if mimics are CAPABLE of jumping), but maybe there's that could be explored with this.

I haven't worked specifically in multiplayer games, and I'm much more akin to Unreal then Unity, but I think for this idea to be possible would require mimics to be converted into actual player characters with mimic abilities rather than trying to do it the other way around. I know there's the mod "Lethal Internship" that has AI with able to complete player tasks, but again I'm not sure what extents that mod had to take.

(Also another thing not exactly related to this idea, my friends and I are always able to determine if a mimic IS a mimic based solely on the fact that they never have an item in their hands. Again, don't know how easy it would be to implement, but having mimics able to hold objects, and then drop them when they're killed, would give them much more believably and may also offer players an extra reason to seek out fighting mimics.)

I appreciate the work that you've done with this mod, it has made the game exponentially more fun for my friends and I, so I thought I'd throw out some ideas in case you were looking to expand it more in the future.

**Comment by qwbarch** [comment]
Sorry for the late response. At one point another dev and I were planning on making a huge overhaul to improve Mirage, which included masked behavioral changes, but we no longer have the time to dedicate to the mod anymore.
Mirage is basically in maintenance mode, where I'll try to fix bugs here and there, but I probably won't be working on any new feature requests anymore.

Glad you like the mod though!
